internal/server/singleprocess/state/track_task.go

```diff
@@ -0,0 +1,317 @@
+package state
+
+import (
+	"strings"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/hashicorp/go-memdb"
+	bolt "go.etcd.io/bbolt"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	pb "github.com/hashicorp/waypoint/pkg/server/gen"
+)
+
+var tracktaskBucket = []byte("tracktask")
```

Comment:
Bikeshed for the whole thing: I'd name this `task`. State by definition is tracking something, so i.e. we don't do `trackjob`, just `job`. Similar vibe here.

Reply:
Totally - I initially reached for calling it `Task`, and wondered since it's a top level message maybe `Task` is better reserved for something else, but this is probably the thing it should be!

Reply:
Done!